### ISSUE TITLE
[Spike]: Implement duplicated payment methods detection mechanism

### ISSUE THREAD

**Issue by timur27** [woothemes/woocommerce#45218]
### Describe the solution you'd like

Recently, we explored technical approaches to identify duplicate payment methods enabled by plugins on a store. Two solutions were proposed, the simplest one and a more sophisticated one. The simplest solution is based on preparing the hardcoded list of known payment method names used across different plugins and check if the payment method name matches the list multiple times. The more sophisticated approach bases on defining the new contract for plugins to follow by extending the main gateway class.

There was also discussion about where this feature needs to be placed. The idea comes from the Transact group, that's why WooPayments was often considered as a potential candidate for the future to be implemented.

Finally, it was concluded that the feature should be developed in two steps. First, the simplest solution will be built in WooPayments. Once the feature impact is proven, the sophisticated approach should be implemented and it should be done in WooCommerce. **This issue is to implement the more sophisticated approach for duplicated payment methods detection mechanism in WooCommerce.**

### Additional context
* paJDYF-ck0-p2 represents the P2 post, containing both solutions
* the spike PR containing drafts for both solutions is https://github.com/woocommerce/woocommerce/pull/44617
* main discussion threads are in paJDYF-ck0-p2#comment-23314 and paJDYF-ck0-p2#comment-23652 

#### Acceptance criteria

> 💡 All the classes and implementation details below are related to https://github.com/woocommerce/woocommerce/pull/44617

1. Duplicate detection mechanism is developed further in https://github.com/woocommerce/woocommerce/pull/44617, or a separate PR is created, which takes the most relevant pieces from the spike implementation.
2. Duplicates finding implementation from `WC_Payment_Method_Type_Duplicates_Finder` is modified further and reaches production readiness.
3. Checkbox to turn the feature on and off is implemented and functional
4. The detection mechanism output is not used anywhere just yet
5. The Core team is asked about the best practices of releasing this feature and in case anything is actionable in code (e.g. create a warning that would inform plugin developers about the new field in the gateway object), this is implemented as well.

**Comment by timur27** [comment]
@pierorocca I have a question about 

> 3. Checkbox to turn the feature on and off is implemented and functional

Do we have any suggestions on where this should be placed within Woo? My first idea was to have it under `WooCommerce -> Settings -> Payments`. If so, some placement/design suggestions would be highly appreciated.

**Comment by pierorocca** [comment]
Yes have some ideas and questions. Posted to the Slack conversation. Thanks for raising these important questions.

**Comment by gpressutto5** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/woocommerce/woocommerce/45218?planning-poker) with Zenhub @FangedParakeet

**Comment by flakas** [comment]
Closing this issue as the Payment Registration API supersedes this work.
